ggv_bin.cc

```diff
@@ -86,6 +88,8 @@ static void
 ggv_bin_read_text32(QDataStream& stream, QByteArray& buf, const char* descr = nullptr)
 {
   quint32 len = ggv_bin_read32(stream, descr);
+  if (len > INT_MAX)
```

Comment:
Should we push this test into read32() itself? 
This probably fails in ILP64, but I don't know if any such architectures remain...

Are there any real-world cases where these conditions can happen and we could do better than a fatal() or is this basically damaged files that we'd be reading?

Reply:
I wouldn't push the test into read32() as there are other places that use this function for values where the meaning and value range is not understood (and can be ignore for the purpose of extracting the geo data). Hence we might reject legitimate files if we push the check into read32().

You are right about the ILP64, this should rather be INT32_MAX. I will update the diff.

This check is added to catch damaged files. There are no real world uses cases. None of my 150 test samples triggers this case. So I have created a test case manually to check what would happen and noticed that there's only a warning:

QIODevice::read (QFile, "overflow.ovl"): Called with maxSize < 0

But no error status is returned, parsing continues at a wrong offset and fails shortly after one of the next values is rejected because it is out of range.